modules/andes-core/common/src/main/java/org/wso2/andes/security/AMQPSecurityUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.wso2.andes.security;
+
+import org.wso2.andes.AMQProtocolException;
+
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Utility class for AMQP security functions.
+ */
+public class AMQPSecurityUtils {
+
+    public static final String RESPONSE_CHARSET = "UTF8";
+    private static byte SEPARATOR = 0;
+
+    public static byte[] encodeCredentials(String authenticationID, String password) throws AMQProtocolException {
+        try {
+            byte[] passwordBytes = password.getBytes(RESPONSE_CHARSET);
+            byte[] authnid = authenticationID.getBytes(RESPONSE_CHARSET);
+            byte[] encodedCredentials = new byte[passwordBytes.length + authnid.length + 2];
+            int size = 0;
+
+            encodedCredentials[size++] = SEPARATOR;
```

Comment:
would it be worthwhile to explain what is happening here.. (use of size variable)?